closed/src/java.base/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -32,17 +32,28 @@
 import jdk.internal.reflect.Reflection;
 import jdk.internal.reflect.CallerSensitive;
 
+import sun.security.action.GetPropertyAction;
+
 public class NativeCrypto {
 
+    private static final String nativeCryptTrace = GetPropertyAction.privilegedGetProperty("jdk.nativeCryptoTrace");
     private static final boolean loaded = AccessController.doPrivileged(
             (PrivilegedAction<Boolean>) () -> {
             Boolean isLoaded = Boolean.FALSE;
             try {
                 System.loadLibrary("jncrypto"); // check for native library
                 // load OpenSSL crypto library dynamically.
-                if (loadCrypto() == 0) {
+                StringBuffer dataBuffer = new StringBuffer();
+                if (loadCrypto(dataBuffer) == 0) {
```

Comment:
Already changed to pass a boolean for the cryptoTrace setting rather than passing the StringBuffer.  This allows the C code to print the trace and not have to pass the trace string back to Java to print.